### DISCUSSION TITLE
Feature Request: MySQL Scanner Extension: select data from mysql table partition

### DISCUSSION THREAD

**Discussion by thomas-0816** [duckdb/duckdb#9650] [Ideas]
MySQL supports partitioning of tables.
(source https://dev.mysql.com/doc/refman/8.0/en/partitioning-selection.html)

e.g. to read data from a specific partition, the following select syntax is used:
```
SELECT * FROM employees PARTITION (p1);
SELECT * FROM employees PARTITION (p0, p2);
```

Similar to "truncate table employees", mysql also supports a very fast delete operation for truncating partitions:
```
ALTER TABLE employees TRUNCATE PARTITION p0;
ALTER TABLE employees TRUNCATE PARTITION p1, p3;
```

This makes partitions a good choice when exporting data to other systems - like duckdb - since "delete from ..." is a very slow operation with big tables in mysql.

Therefore it would be great if the new MySQL Scanner Extension would support mysql partitions as well.

Looking at https://duckdb.org/docs/extensions/mysql_scanner I'm not sure if this feature is already implemented in the preview or if it is a new feature request.

**Comment by Mause** [comment]
The MySQL extension exists here: https://github.com/duckdb/duckdb_mysql

Feel free to raise that feature request there

**Reply by thomas-0816** [reply]
Currently the extension repository does not yet have discussions enabled, so I would not file this as a bug issue.
I will move this request as soon as possible.
